### PR TITLE
Add examples for async iterators and generators.

### PR DESCRIPTION
**Pull request by jpmedley**
https://www.chromestatus.com/feature/5727385018171392

### CODE REVIEW

**Comment by jpmedley** [comment]
Is there anything I can do to move this forward?

**Comment by chrisdavidmills** [comment]
@wbamberg — flagging this one.

**Comment by schalkneethling** [comment]
Hey @jpmedley Thanks for your pull request. Apologies for how long it has taken for someone to take a look at it. Couple of items

You will need to add the following entries to the `meta.json` file inside `live-examples/js-examples/statement`. This is so that the relevant pages for the examples are built:

```
...
"statementAsyncForIn": {
    "baseTmpl": "tmpl/live-js-tmpl.html",
    "exampleCode":
        "live-examples/js-examples/statement/statement-async-for-in.html",
    "fileName": "statement-asyncforin.html",
    "title": "JavaScript Demo: Statement - Async For In",
    "type": "js"
},
"statementAsyncFunctionAsterisk": {
    "baseTmpl": "tmpl/live-js-tmpl.html",
    "exampleCode":
        "live-examples/js-examples/statement/statement-async-function-asterisk.html",
    "fileName": "statement-asyncfunctionasterisk.html",
    "title": "JavaScript Demo: Statement - Async Function Asterisk",
    "type": "js"
},
...
```

The following the general file naming conventions, and specifically the one's in this folder. I would change the filenames to the following:

```
statement-async-for-in.html
statement-async-function-asterisk.html
```

With the above you can test these examples locally by running `npm start` from the terminal, and then pointing your browser at `http://localhost:9090/pages/js/`

For the for-in example specifically, I get the following output when running the example:

```
> "Regular iterator"
> [object Promise]
> [object Promise]
> [object Promise]
> "Async iterator"
> 0.7360924091
> 0.7858652753
> 0.5627363239
```

Perhaps that is the intended output? I would recommend adding a comment to each `console.log` that demonstrates the expected output to the end user. Thanks again for your contribution. o/\o

**Comment by jpmedley** [comment]
PTAL

**Comment by jpmedley** [comment]
Now PTAL.

File: live-examples/js-examples/statement/statement-async-function-asterisk.html [hunk 1]
```diff
@@ -0,0 +1,22 @@
+<pre>
+<code id="static-js" data-height="taller">async function* randomNumbers() {
+  const url = 'https://www.random.org/decimal-fractions/?num=1&dec=10&col=1&format=plain&rnd=new';
+
+  console.log("Print randomly generated numbers until reset is clicked the random number exceeds 0.95.");
```
Comment (schalkneethling) [inline live-examples/js-examples/statement/statement-async-function-asterisk.html]: Looks to be a small typo here:

`"Print randomly generated numbers until reset is clicked the random number exceeds 0.95."`

to

`"Print randomly generated numbers until reset is clicked, (or) the random number exceeds 0.95."`

Comment (schalkneethling) [reply live-examples/js-examples/statement/statement-async-function-asterisk.html]: @jpmedley No worries, let me fix that one, and merge. Thanks again for your contribution.

**Pull request merged**